subsys/bluetooth/host/att.c

```diff
@@ -2880,6 +2897,19 @@ static int bt_att_recv(struct bt_l2cap_chan *chan, struct net_buf *buf)
 		}
 	}
 
+	if (handler->type == ATT_REQUEST && !att_chan->transaction_buf) {
+		/* Re-use the REQ buffer (that comes from HCI).
+		 *
+		 * This allows ATT to always be able to send a RSP (or err RSP)
+		 * to the peer, regardless of the TX buffer usage by other stack
+		 * users (e.g. GATT notifications, L2CAP using global pool, SMP,
+		 * etc..), avoiding an ATT timeout due to resource usage.
+		 *
+		 * The ref is taken by `bt_att_chan_create_pdu`.
+		 */
+		att_chan->transaction_buf = buf;
+	}
+
```

Comment:
att_chan->transaction_buf = NULL;
before exiting this function, since it "borrows" from this function scope.